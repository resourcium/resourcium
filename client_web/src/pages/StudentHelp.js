import React, { useMemo } from 'react';
import Tabs from 'src/components/Tabs';
import MSLearnContainer from 'src/components/Embeds/MSLearnContainer';
import LinkedinLearningContainer from 'src/components/Embeds/LinkedinLearningContainer';
import Chatbotv1 from 'src/components/Embeds/Chatbot';
import { QNA_BOT_TOKEN } from 'src/common/config';

function StudentHelpTabs() {
    const tabs = useMemo(() => {
        const MSLearn = {
            content: <MSLearnContainer />,
            label: 'Microsoft Learn',
        };

        const LinkedinLearning = {
            content: <LinkedinLearningContainer />,
            label: 'Linkedin Learning',
        };

        const Chatbot = QNA_BOT_TOKEN ? {
            content: <Chatbotv1 />,
            label: 'QnABot V1',
        } : null;
        return { MSLearn, LinkedinLearning, Chatbot };
    }, []);

    return <Tabs tabs={tabs} />;
}

export default function StudentHelp() {
    return (
        <>
            <div className="StudentHelp page">
                <h2>Student Help page</h2>
                <StudentHelpTabs />
            </div>
        </>
    );
}
