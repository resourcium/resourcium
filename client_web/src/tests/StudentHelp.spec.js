import { create } from 'react-test-renderer';
import React from 'react';

import StudentHelp from 'src/pages/StudentHelp';

import { wrapProvider } from 'src/tests/util';

test('greetings snapshot matches', () => {
    expect(create(wrapProvider(<StudentHelp />))).toMatchSnapshot();
});