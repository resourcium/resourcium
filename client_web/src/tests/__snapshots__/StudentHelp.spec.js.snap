// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`greetings snapshot matches 1`] = `
<div
  className="StudentHelp page"
>
  <h2>
    Student Help page
  </h2>
  <div
    className="tabs"
  >
    <div
      className="tabs-label-row"
    >
      <div
        className="tabs-label selected"
        onClick={[Function]}
      >
        Microsoft Learn
      </div>
      <div
        className="tabs-label"
        onClick={[Function]}
      >
        Linkedin Learning
      </div>
      <div
        className="tabs-label"
        onClick={[Function]}
      >
        QnABot V1
      </div>
    </div>
    <div
      className="tabs-content"
    >
      <p>
        Loading modules...
      </p>
    </div>
  </div>
</div>
`;
